### ISSUE TITLE
`connectAsync` is never resolved with `manualConnect: true`

### ISSUE THREAD

**Issue by dunielpls** [mqttjs/MQTT.js#1661]
Hello,

I am using MQTT.js for a project of mine and discovered that the `connectAsync` function never resolves when it is called with the option `manualConnect` set to `true`.

Example:
```js
import mqtt from "mqtt";

console.log("before connectAsync()"); // obviously works
const client = await mqtt.connectAsync("mqtt://127.0.0.1:1883", {
    manualConnect: true,
});
console.log("after connectAsync()"); // never called
```

I believe this is because the promise that is returned never gets resolved nor rejected because it can only be resolved by firing `"connect"` or `"end"` events that never fire with `manualConnect: true` ref https://github.com/mqttjs/MQTT.js/blob/main/src/lib/connect/index.ts#L195

I suggest adding code to instantly return `Promise.resolved(client)` if `manualConnect` is `true` and adding a check to the current `resolve`/`reject` code to avoid resolving or rejecting the promise multiple times.

It is possible that I will open a PR myself given I have available time, but regardless, I am posting this issue to make the problem known.

**Comment by robertsLando** [comment]
Because calling `connectAsync` with `manualConnect` just makes no sense... just use normal connect with `manualConnect` set to true so you will have a MqttClient instance, then you can:

```js
const client = connect({manualConnect: true})

client.connect()

await once(client, 'connect')
```

**Comment by dunielpls** [comment]
I did just that to solve the issue and wrapped the client in a resolved promise because I needed to include creating a client in other code oriented around promise APIs. I was however stumped for a bit because the documentation did not mention that fulfilling the promise was dependent on the `'connect'` event.

**Comment by robertsLando** [comment]
The connect async returns when the client is connected that's the entire point of it. It's ported from the `async-mqtt` library, it worked like that even there

**Comment by dunielpls** [comment]
I was using it because it is the only care-free way of constructing a `Client` instance, and I was confused because the `connect` and `connectAsync` methods are supposed to behave similarly (according to the documentation), but that was not the case.

Either way, the problem will be solved with the future change in API you mentioned. Thank you for the quick response as well.

**Comment by robertsLando** [comment]
I agree the docs may be more clear about this. Feel free to open a PR to fix the docs if possible, thanks 🙏🏼
